### ISSUE TITLE
Add support to install a specific version of extension

### ISSUE THREAD

**Issue by devaradhanm** [Azure/azure-cli-extensions#1587]
### Description of issue (in as much detail as possible)
This could be a duplicate of https://github.com/Azure/azure-cli-extensions/issues/1123
When extension gets updated & have blocking issues, there is no way for the customers to use the older version & hence blocks the deployments.

For eg, we were blocked due to this issue https://github.com/Azure/azure-cli-extensions/issues/1578 till a fix was available. 

If the "az extension add" command supports pinning to a particular version, we would have been able to unblock ourselves. (similar to using specific az cli version if we face any issues with the latest version)

-----

**Comment by gusnaughton** [comment]
From other thread:
It seems like currently the only way to achieve this is to store the desired commit SHA alongside your code's version, and at build time pulling down this extensions repository against that SHA by using the [--source parameter on az extension add](https://docs.microsoft.com/en-us/cli/azure/extension?view=azure-cli-latest#az-extension-add) to add the versioned extension. 

This adds significant complexity to build pipelines. It would also be great if a flag was introduced to allow pulling extensions from private registry should an organization elect to fork an extension.

**Comment by yonzhan** [comment]
add to S170

**Comment by haroldrandom** [comment]
@gusnaughton I think it's a feature to request to support Azure CLI resolve/install extension from a private registry.
Could you create another issue to track it? I will fix the original one here.

**Comment by gusnaughton** [comment]
> @gusnaughton I think it's a feature to request to support Azure CLI resolve/install extension from a private registry.
> Could you create another issue to track it? It might be complex to fix all those ones together.
> It would be better to solve those requests step by step.

Sure thing, Harold. https://github.com/Azure/azure-cli-extensions/issues/1836

**Comment by haroldrandom** [comment]
@devaradhanm CLI now support install extension with specific version. This feature is going to ship in Azure CLI 2.8.0 and expected at 6/23/2020. 

If it's urgent, you could try it against dev branch or edge build
